packages/atomic/src/components/result-lists/result-list-common.tsx

```diff
@@ -0,0 +1,316 @@
+        },
+      };
+    }
+
+    this.registerResultTemplates(
+      opts.templateElements,
+      opts.includeDefaultTemplate,
+      opts.onReady,
+      opts.onError
+    );
+  }
+
+  set renderingFunction(render: ResultRenderingFunction) {
+    this.render = render;
+  }
+
+  private determineAllFieldsToInclude(
+    configuredFieldsToInclude: string
+  ): string[] {
+    if (configuredFieldsToInclude.trim() === '')
+      return [...EcommerceDefaultFieldsToInclude];
+    return EcommerceDefaultFieldsToInclude.concat(
+      configuredFieldsToInclude.split(',').map((field) => field.trim())
+    );
+  }
+
+  public getTemplate(result: Result) {
+    return this.resultTemplatesManager.selectTemplate(result);
+  }
+
+  public getContentOfResultTemplate(
+    resultOrFolded: Result | FoldedResult
+  ): HTMLElement | DocumentFragment {
+    const result = (resultOrFolded as FoldedResult).result || resultOrFolded;
+    return this.render ? this.render(result) : this.getTemplate(result)!;
+  }
+
+  private makeDefaultTemplate(): ResultTemplate<DocumentFragment> {
+    const content = document.createDocumentFragment();
+    const linkEl = document.createElement('atomic-result-link');
+    content.appendChild(linkEl);
+    return {
+      content,
+      conditions: [],
+    };
+  }
+
+  public async registerResultTemplates(
+    elements: NodeListOf<TemplateElement>,
+    includeDefaultTemplate = true,
+    onReady: () => void,
+    onError: () => void
+  ) {
+    this.resultTemplatesManager = buildResultTemplatesManager(
+      this.bindings.engine
+    );
+
+    const customTemplates = await Promise.all(
+      Array.from(elements).map(async (resultTemplateElement) => {
+        const template = await resultTemplateElement.getTemplate();
+        if (!template) {
+          onError();
+        }
+        return template;
+      })
+    );
+    const templates = (
+      includeDefaultTemplate ? [this.makeDefaultTemplate()] : []
+    ).concat(
+      customTemplates.filter(identity) as ResultTemplate<DocumentFragment>[]
+    );
+
+    this.resultTemplatesManager.registerTemplates(...templates);
+    onReady();
+  }
+
+  public getResultId(
+    result: Result | FoldedCollection,
+    resultListState: FoldedResultListState | ResultListState
+  ) {
+    return (
+      this.getUnfoldedResult(result).uniqueId + resultListState.searchResponseId
+    );
+  }
+
+  public getUnfoldedResult(result: Result | FoldedResult): Result {
+    return (result as FoldedResult).result ?? result;
+  }
+
+  public handleInfiniteScroll(
+    isEnabled: boolean,
+    host: HTMLElement,
+    resultList: ResultList | FoldedResultList
+  ) {
+    if (!isEnabled) {
+      return;
+    }
+
+    const hasReachedEndOfElement =
+      window.innerHeight + window.scrollY >= host.offsetHeight;
+
+    if (hasReachedEndOfElement) {
+      resultList.fetchMoreResults();
+    }
+  }
+
+  public componentDidRender(
+    resultListState: FoldedResultListState | ResultListState,
```

Comment:
Similarly to below, instead of taking the state here, I think we should take `firstSearchExecuted` directly.

Reply:
yeah, that makes sense